### DISCUSSION TITLE
Will creating persistent data in a submod potentially break MAS?

### DISCUSSION THREAD

**Discussion by Wet2083** [Monika-After-Story/MonikaModDev#7768] [QA - code and content creation (technical)]
Hi all, i'm working on a submod currently that requires Monika to remember choices the Player makes. I was wondering if creating persistent data in a submod could potentially break MAS if it was uninstalled. Thanks in advance!

**Comment by lunulae** [comment]
If we're just talking player variables (like remembering if they like mint ice cream, for example) it will not break MAS if it's uninstalled. 

However, you should try to name the persistent variable that you're creating in a way that is unique or indicates your specific submod, so that if the core game or another submod has a variable of the same kind it won't accidentally find one by the same name and create a conflict.

**Comment by Booplicate** [comment]
It may if you save data of a non-builtin type. If it's just a boolean (`True`/`False`) or other builtin types, then it's probably fine. But you should name your variables accordingly so they don't interfere with ours or other submods vars. Ideally give it a prefix like `_yoursubmodsprefix_variable_name`.
